services/src/main/java/org/keycloak/protocol/saml/SamlService.java

```diff
@@ -553,7 +553,8 @@ protected Response logoutRequest(LogoutRequestType logoutRequest, ClientModel cl
 
                     //artifact binding state must be attached to the user session upon logout, as authenticated session
                     //no longer exists when the LogoutResponse message is sent
-                    if ("true".equals(clientSession.getNote(JBossSAMLURIConstants.SAML_HTTP_ARTIFACT_BINDING.get()))){
+                    if ("true".equals(clientSession.getNote(JBossSAMLURIConstants.SAML_HTTP_ARTIFACT_BINDING.get()))
+                            && samlClient.useArtifactBindingLogout()) {
                         userSession.setNote(JBossSAMLURIConstants.SAML_HTTP_ARTIFACT_BINDING.get(), "true");
```

Comment:
So if SamlArtifactBindingLogout is disabled, you don't set the user session note... And what happens? It acts as a standard saml or it does nothing? (I guess it is the second option)

Reply:
It does nothing particular, thus it falls back on the standard use case --> use POST / Redirect binding according to the client configuration.
Cf. https://github.com/cloudtrust/keycloak/blob/a20f3b726a7f815f14cdf11e4aea1581dfe453fe/services/src/main/java/org/keycloak/protocol/saml/SamlProtocol.java#L609